File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {

        Set<Integer> set = new HashSet<>();
        
        for(int i=0; i<nums.length; i++) {
            set.add(nums[i]);
        }
        
        int result = Math.min(set.size(), nums.length/2);
        
        return result;
    }
}

File: Python3/프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    
    count = {}
    for i in nums:
        if i in count:
            count[i] += 1
        else:
            count[i] = 1
    
    if len(count) >= len(nums) / 2:
        return len(nums) / 2
    else:
        return len(count)